tests/utils/proc_mock.py

```diff
@@ -88,8 +90,27 @@ def should_bad_exit_on(self, cmd: list[str] | str, exit_code: int = -1, output:
             mock_data.output_lines = output
         self._add_mock_data(cmd, mock_data)
 
-    def set_output(self, cmd: list[str] | str, output: list[str]) -> None:
-        self._add_mock_data(cmd, CommandMockData(output_lines=output))
+    def set_output(
+        self,
+        cmd: list[str] | str,
+        output: list[str],
+        side_effect: Callable[[Any], None] | None = None,
+        side_effect_args: dict[str, Any] | None = None,
+    ) -> None:
+        """
+        Set the output of a command, and optionally a side effect to be called when the command is run.
+        Side effect can't have input arguments, the main utility is running extra logic on top of execution of the
```

Comment:
Change the wording here, side effect can have input arguments now so probably just explain what side effect is (allows you to execute some custom function during execution of a mock proc_mock. But the sentenc saying it can't have input args should be removed as we pass the side_effect_args now